ui/lesson/OpRunner/OpRunner.tsx

```diff
@@ -65,6 +65,85 @@ const OpRunner = ({
   const { ref: arrowContainerRef } = useArrows()
   const scrollPosition = useHorizontalScroll(scrollRef)
 
+  const [executor, setExecutor] = useState<LanguageExecutor | null>(null)
+
+  const initializeExecutor = () => {
+    const initialStackArray = initialStack
+      .split(' ')
+      .filter((item) => item.trim())
+    const tokens = LanguageExecutor.parsableInputToTokens(
+      LanguageExecutor.rawInputToParsableInput(script)
+    )
+    const newExecutor = new LanguageExecutor(tokens, initialStackArray, height)
+    setExecutor(newExecutor)
+
+    // Create an initial state to represent the initial stack
+    const initialState = {
+      stack: initialStackArray,
+      operation: {
+        tokenType: null,
+        resolves: null,
+        value: 'INITIAL_STACK',
+        type: '',
+      },
+      step: -1, // Use -1 to indicate this is the initial state
+      negate: 0,
+      error: null,
+    }
+
+    setStackHistory([initialState])
+  }
+
+  const handleStep = () => {
+    if (executor) {
+      const state = executor.executeStep()
+      if (state) {
+        setStackHistory((prev) => [...prev, state])
+        checkSuccessState(executor.tokens, state.stack)
+      }
+    } else {
+      initializeExecutor()
+    }
+  }
```

Comment:
If we could make this behave like Run but on an async loop it would be nice to not have to continually click the step button over and over